mlir/lib/Driver/CompilerDriver.cpp

```diff
@@ -634,19 +694,27 @@ LogicalResult QuantumDriverMain(const CompilerOptions &options, CompilerOutput &
         llvmModule->setDataLayout(targetMachine->createDataLayout());
         llvmModule->setTargetTriple(targetTriple);
 
-        if (failed(timer::timer(runLLVMPasses, "runLLVMPasses", /* add_endl */ false, options,
-                                llvmModule, output))) {
-            return failure();
-        }
-
         catalyst::utils::LinesCount::Module(*llvmModule.get());
 
-        if (failed(timer::timer(runEnzymePasses, "runEnzymePasses", /* add_endl */ false, options,
-                                llvmModule, output))) {
-            return failure();
+        if (options.asyncQnodes) {
+            if (failed(timer::timer(runCoroLLVMPasses, "runCoroLLVMPasses", /* add_endl */ false,
+                                    options, llvmModule, output))) {
+                return failure();
+            }
```

Comment:
Adding the instrumentation here as well:
```suggestion
            }
            catalyst::utils::LinesCount::Module(*llvmModule.get());
```